canvas_modules/common-canvas/src/common-properties/controls/expression/expression-builder/expression-builder.jsx

```diff
@@ -31,38 +31,39 @@ export default class ExpressionBuilder extends React.Component {
 		this.onChange = this.onChange.bind(this);
 		this.onBlur = this.onBlur.bind(this);
 		this.onSelectionChange = this.onSelectionChange.bind(this);
+		this.getCodemirrorState = this.getCodemirrorState.bind(this);
 	}
 
 	onChange(newValue) {
 		const value = (typeof newValue === "string") ? newValue : newValue.toString();
-		const somethingSelected = this.editor.viewState.state.selection.ranges.some((r) => !r.empty);
-		let cursor = this.editor.viewState.state.selection.main.head;
+		const somethingSelected = this.getCodemirrorState()?.selection.ranges.some((r) => !r.empty);
+		let cursor = this.getCodemirrorState()?.selection.main.head;
```

Comment:
This doesn't really help protect us. It would have to be `this.getCodemirrorState()?.selection?.main?.head`

Anything that has methods would have to change the object first, then call the method.  

Is everything your using documented?  If so, maybe we're safe to use it without optional chaining.

Reply:
Everything after `this.getCodemirrorState()?` is documented in [migration guide](https://codemirror.net/docs/migration/) and [reference manual](https://codemirror.net/docs/ref/).